File: src/main/java/com/dozycoffee/wms/warehouse/application/port/in/RegisterWarehouseUseCase.java
package com.dozycoffee.wms.warehouse.application.port.in;

import com.dozycoffee.wms.warehouse.application.port.in.command.RegisterWarehouseCommand;
import com.dozycoffee.wms.warehouse.application.port.in.result.WarehouseResult;
import reactor.core.publisher.Mono;

public interface RegisterWarehouseUseCase {

    Mono<WarehouseResult> register(RegisterWarehouseCommand command);
}

File: src/main/kotlin/com/dozycoffee/wms/warehouse/application/port/in/RegisterWarehouseUseCase.kt
package com.dozycoffee.wms.warehouse.application.port.`in`

import com.dozycoffee.wms.warehouse.application.port.`in`.command.RegisterWarehouseCommand
import com.dozycoffee.wms.warehouse.application.port.`in`.result.WarehouseResult
import reactor.core.publisher.Mono

interface RegisterWarehouseUseCase {
    fun register(command: RegisterWarehouseCommand): Mono<WarehouseResult>
}
